### PR TITLE
SIL: Remove special meaning for @_semantics("stdlib_binary_only")

### PR DESCRIPTION
**Pull request by slavapestov**
With -sil-serialize-all gone, this no longer means anything; just
don't declare the function as @_inlineable instead.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swiftix @moiseev Let me know if this makes sense.

**Comment by slavapestov** [comment]
Also @milseman for the string stuff

**Comment by slavapestov** [comment]
I haven't even built this locally, so of course all tests will pass!

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please clean smoke test Linux

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke benchmark

**Comment by slavapestov** [comment]
@swift-ci Please smoke benchmark

File: docs/HighLevelSILOptimizations.rst [hunk 1]
```diff
@@ -117,15 +117,13 @@ Cloning code from the standard library
...
-application. This allows the optimizer to inline calls from stdlib and improve
-the performance of code that uses common operators such as '++' or basic
-containers such as Array. However, importing code from the standard library can
-increase the binary size. Marking functions with @_semantics("stdlib_binary_only")
-will prevent the copying of the marked function from the standard library into the
-user program.
-
-Notice that this annotation is similar to the resilient annotation that will
-disallow the cloning of code into the user application.
+application for functions marked @_inlinable. This allows the optimizer to
```
Comment (moiseev) [inline docs/HighLevelSILOptimizations.rst]: s/`inlinable`/`inlineable` It's not codable =)

File: docs/HighLevelSILOptimizations.rst [hunk 2]
```diff
@@ -117,15 +117,13 @@ Cloning code from the standard library
...
+inline calls from the stdlib and improve the performance of code that uses
+common operators such as '++' or basic containers such as Array. However,
```
Comment (moiseev) [inline docs/HighLevelSILOptimizations.rst]: I know it's not your words, but you touched it, so... `++` does not exit any more.

File: test/SILOptimizer/Inputs/linker_pass_input.swift [hunk 3]
```diff
@@ -2,23 +2,26 @@
...
   unknown()
 }
 
-struct A {}
+@_versioned struct A {
+  @_versioned init() {}
+}
+
 @inline(never)
+@_inlineable
```
Comment (moiseev) [inline test/SILOptimizer/Inputs/linker_pass_input.swift]: Wait, I thought `@inline(never)` and `@_inlineable` don't play well together, and already wanted to ask for a warning.

Comment (milseman) [reply test/SILOptimizer/Inputs/linker_pass_input.swift]: `@_inlineable` is not about inlining, so they are actually orthogonal.

Comment (moiseev) [reply test/SILOptimizer/Inputs/linker_pass_input.swift]: But, @swiftix told me that `@inline(__always)` implies `@_inlineable`. I'm confused now...

Comment (swiftix) [reply test/SILOptimizer/Inputs/linker_pass_input.swift]: @moiseev IIRC, the current rules in the compiler work in such a way that  `@inline(__always)` and `@_transparent` implies @_inlineable. 

@slavapestov That's how it works today, or?

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/StringBridge.swift [hunk 4]
```diff
@@ -47,9 +47,8 @@ func _stdlib_binary_CFStringGetCharactersPtr(
 
 /// Bridges `source` to `Swift.String`, assuming that `source` has non-ASCII
 /// characters (does not apply ASCII optimizations).
-@_inlineable // FIXME(sil-serialize-all)
 @_versioned // FIXME(sil-serialize-all)
-@inline(never) @_semantics("stdlib_binary_only") // Hide the CF dependency
+@inline(never) // Hide the CF dependency
```
Comment (milseman) [inline stdlib/public/core/StringBridge.swift]: IIRC The comment is solely from the lack of `@_inlineable`. I don't think it goes on `@inline(never)`, but I don't know where to put it. Perhaps under the new paradigm it's just not an interesting distinction. Thoughts?

**Review by milseman — COMMENTED** [review]
Question about the use of decl comments and convention. The String file changes LGTM.

**Comment by slavapestov** [comment]
@swift-ci Please clean smoke test Linux

File: stdlib/public/core/StringBridge.swift (hunk 4 above)
Comment (slavapestov) [reply stdlib/public/core/StringBridge.swift]: I think the comment is still useful for future maintainers in case they decide to put @_inlineable back in?

**Comment by jrose-apple** [comment]
I'm not sure why testing didn't catch this, but StdlibUnittest is still using this. Reverting for now.

File: stdlib/public/core/StringBridge.swift (hunk 4 above)
Comment (jrose-apple) [reply stdlib/public/core/StringBridge.swift]: In theory we shouldn't need the inline-never anymore either, right?

Comment (swiftix) [reply stdlib/public/core/StringBridge.swift]: Without inline-never it can be inlined inside the stdlib itself.

Comment (jrose-apple) [reply stdlib/public/core/StringBridge.swift]: Right, but that should be fine now.

Comment (milseman) [reply stdlib/public/core/StringBridge.swift]: Outline-and-inline-never is still a useful, albeit niche, tool inside the stdlib when Builtin.expect is insufficient.

**Pull request merged**